### PR TITLE
Adds 23 byte protocol for Firmware 11 (AK001-ZJ21411)

### PR DESCRIPTION
**Pull request by kirillsimin**
This update adds a new 23 byte protocol for Zengge / Flux Led bulb Firmware 11.x, model 0x35, AK001-ZJ21411. The signature has changed completely, with some parts of the sequence that I don't fully understand yet. This PR has working **RGB** and **CW** logic. It doesn't have the transition logic yet because more deciphering of the byte sequence is needed for that.

❗Please pay close attention to the following change in `base_device.py`, line `1137` and changes caused by it:
```
level_write_mode = self._protocol.get_write_mode()
```
The new bulb has a different method of setting color / white write mode, and so I put the logic for setting it into the base protocol class, rather than in a constant.

❗Another deep change is making `increment_counter` a public function, since I believe this bulb accepts a counter as its 7th byte. It's not necessary for it to function, so if this is an unwanted change, it could easily be reversed.

### CODE REVIEW

**Comment by bdraco** [comment]
@kirillsimin I think this nearly is ready to go if I didn't break anything in the refactoring. The only thing missing is the test coverage for the `if red is not None and green is not None and blue is not None:` and `elif cool_white is not None:`

**Comment by bdraco** [comment]
I’m traveling today and tomorrow so I’ll take a look when I get back home. I bought some of these devices to test with so hopefully they show up tomorrow

**Comment by kirillsimin** [comment]
Fantastic. Thanks so much @bdraco . I've been testing locally with the 9w surplife version.

**Comment by bdraco** [comment]
Test devices I ordered showed up.

Bulb RGBCW (0x35)
by Zengge
Firmware: 11.88
Hardware: AK001-ZJ21411
MAC: 7C:B9:4C:97:3C:F5

The work just fine without any of these changes though so maybe they have changed the firmware to accept the old commands as well?

**Comment by bdraco** [comment]
I did some testing with this PR. Everything works great except for setting the cool white LEDs. When I set it to 6500 K, the bulb turns off

**Comment by bdraco** [comment]
```
Mar 11 04:01:22 homeassistant homeassistant[589]: 2025-03-10 18:01:22.306 DEBUG (MainThread) [flux_led.base_device] 192.168.106.77: _replace_raw_state: {'red': 0, 'green': 0, 'blue': 0, 'warm_white': 0, 'cool_white': 0}
Mar 11 04:01:22 homeassistant homeassistant[589]: 2025-03-10 18:01:22.306 DEBUG (MainThread) [flux_led.aioprotocol] ('192.168.106.77', 5577) => 0xB0 0xB1 0xB2 0xB3 0x00 0x01 0x02 0x37 0x00 0x0E 0xE0 0x01 0x00 0xB1 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x14 0x00 0x00 0xB4 (25)
Mar 11 04:01:23 homeassistant homeassistant[589]: 2025-03-10 18:01:23.283 DEBUG (MainThread) [flux_led.aioprotocol] ('192.168.106.77', 5577) => 0x81 0x8A 0x8B 0x96 (4)
Mar 11 04:01:23 homeassistant homeassistant[589]: 2025-03-10 18:01:23.283 DEBUG (MainThread) [homeassistant.components.flux_led.coordinator] Finished fetching 192.168.106.77 data in 0.000 seconds (success: True)
Mar 11 04:01:23 homeassistant homeassistant[589]: 2025-03-10 18:01:23.373 DEBUG (MainThread) [flux_led.aioprotocol] ('192.168.106.77', 5577) <= 0x81 0x35 0x23 0x61 0x30 0x50 0x00 0x00 0x00 0x00 0x0A 0x00 0x0F 0xD3 (14)
Mar 11 04:01:23 homeassistant homeassistant[589]: 2025-03-10 18:01:23.373 DEBUG (MainThread) [flux_led.base_device] 192.168.106.77: set_available: Received message 813523613050000000000a000fd3
Mar 11 04:01:23 homeassistant homeassistant[589]: 2025-03-10 18:01:23.373 DEBUG (MainThread) [flux_led.base_device] 192.168.106.77: State: LEDENETRawState(head=129, model_num=53, power_state=35, preset_pattern=97, mode=48, speed=80, red=0, green=0, blue=0, warm_white=0, version_number=10, cool_white=0, color_mode=15, check_sum=211)
```

**Comment by kirillsimin** [comment]
I'll take a look at the white color bug, but I'm really surprised that it worked for you on 11.88. The one I received is: 

Bulb RGBCW (0x35)
by Zengge
Firmware: 11.87
Hardware: AK001-ZJ21411

And it only responds to on/off, which is how I found the original issue (#398 ), which mentions firmware 11.94

**Comment by kirillsimin** [comment]
I fixed the bug, but interestingly, when using `-k` arg, the value for `w`, or warm white comes through as `0`, not `None` in:

`base_device.py:1147`

```
        msgs = self._protocol.construct_levels_change(
            persist, r_value, g_value, b_value, w_value, w2_value, write_mode
        )
```

File: flux_led/protocol.py [hunk 1]
```diff
@@ -1378,6 +1403,89 @@ def construct_preset_pattern(
...
+            # warm white comes through as 0, even when temp is set to 6500
+            white_brightness = round(((warm_white + cool_white) / (2 * 255)) * 64)
+            white_temp = round((cool_white / (warm_white + cool_white)) * 64)
+        elif cool_white is not None and warm_white is None:
+            white_temp = 0x64
+            white_brightness = int((cool_white / 255) * 100)
+        else:
+            white_temp = 0x00
+            white_brightness = 0x00
+
```
Comment (bdraco) [inline flux_led/protocol.py:1460]: In the future it might make sense to move some of this into flux_led/utils.py in the event other devices need similar scaling

File: flux_led/protocol.py [hunk 2]
```diff
@@ -1378,6 +1403,89 @@ def construct_preset_pattern(
...
+                        h,
+                        s,
+                        v,
+                        white_temp,
+                        white_brightness,
+                        0x00,
+                        0x00,
+                        0x14,
+                        0x00,
+                        0x00,
```
Comment (bdraco) [inline flux_led/protocol.py:1480]: I was a bit surprised that the checksum is always. 0x00 on the inner message.

I wonder if we can remove `inner_pre_constructed=True` and drop the last byte here

Comment (bdraco) [reply flux_led/protocol.py:1480]: tested, doesn't work

**Comment by bdraco** [comment]
It looks like these send a different message on state change as well

```
2025-03-11 15:56:17.792 DEBUG (MainThread) [flux_led.aioprotocol] ('192.168.106.77', 5577) <= 0xB0 0xB1 0xB2 0xB3 0x00 0x02 0x02 0x69 0x00 0x14 0xEA 0x81 0x01 0x00 0x35 0x0A 0x23 0x61 0x01 0x4F 0x0F 0x00 0x00 0x00 0x64 0x64 0x00 0x00 0x00 0x00 0x9D (31)
2025-03-11 15:56:17.792 DEBUG (MainThread) [flux_led.base_device] 192.168.106.77: set_available: Received message b0b1b2b3000202690014ea810100350a2361014f0f0000006464000000009d
2025-03-11 15:56:17.792 DEBUG (MainThread) [flux_led.aiodevice] 192.168.106.77: Ignoring unknown message: 0xEA 0x81 0x01 0x00 0x35 0x0A 0x23 0x61 0x01 0x4F 0x0F 0x00 0x00 0x00 0x64 0x64 0x00 0x00 0x00 0x00
2025-03-11 15:56:23.873 DEBUG (MainThread) [homeassistant.components.flux_led.coordinator] Finished fetching 192.168.106.77 data in 0.000 seconds (success: True)
2025-03-11 15:56:26.980 DEBUG (MainThread) [flux_led.aioprotocol] ('192.168.106.77', 5577) <= 0xB0 0xB1 0xB2 0xB3 0x00 0x01 0x02 0x21 0x00 0x14 0xEA 0x81 0x01 0x00 0x35 0x0A 0x23 0x61 0x01 0x4F 0x0F 0x00 0x00 0x00 0x64 0x64 0x00 0x00 0x00 0x00 0x54 (31)
2025-03-11 15:56:26.980 DEBUG (MainThread) [flux_led.base_device] 192.168.106.77: set_available: Received message b0b1b2b3000102210014ea810100350a2361014f0f00000064640000000054
2025-03-11 15:56:26.980 DEBUG (MainThread) [flux_led.aiodevice] 192.168.106.77: Ignoring unknown message: 0xEA 0x81 0x01 0x00 0x35 0x0A 0x23 0x61 0x01 0x4F 0x0F 0x00 0x00 0x00 0x64 0x64 0x00 0x00 0x00 0x00
```

**Comment by bdraco** [comment]
I tried to figure the "extended state" out a bit in https://github.com/lightinglibs/flux_led/pull/428 in case that can give you a head start
